File: test/system/focus_test.rb
require "application_system_test_case"

class FocusTest < ApplicationSystemTestCase
  setup do
    visit edit_post_path(posts(:empty))
  end

  test "focus the editor" do
    find_editor.focus
    assert_editor_has_focus
  end

  test "text after focus doesn't add new line" do
    find_editor.focus
    find_editor.send "Hello there"

    assert_editor_html "<p>Hello there</p>"
  end

  test "autofocus attribute" do
    visit edit_post_path(posts(:empty), autofocus: true)
    assert_editor_has_focus
  end

  private
    def assert_editor_has_focus
      assert_equal active_element, find_editor.content_element, "editor should have focus"
    end
end

File: test/browser/tests/focus.test.js
import { test } from "../test_helper.js"
import { expect } from "@playwright/test"
import { assertEditorHtml } from "../helpers/assertions.js"

test.describe("Focus", () => {
  test.beforeEach(async ({ page }) => {
    await page.goto("/posts/new")
    await page.waitForSelector("lexxy-editor[connected]")
  })

  test("focus the editor", async ({ editor }) => {
    await editor.focus()
    await expect(editor.content).toBeFocused()
  })

  test("text after focus doesn't add new line", async ({ editor }) => {
    await editor.focus()
    await editor.send("Hello there")

    await assertEditorHtml(editor, "<p>Hello there</p>")
  })

  test("autofocus attribute", async ({ page, editor }) => {
    await page.goto("/posts/new?autofocus=true")
    await editor.waitForConnected()

    await expect(editor.content).toBeFocused()
  })
})
